src/wavm/threads.cpp

```diff
@@ -286,83 +286,123 @@ I32 s__futex(I32 uaddrPtr,
     return returnValue;
 }
 
-/*
- * --------------------------
- * STUBBED PTHREADS
- * --------------------------
- */
+// --------------------------
+// PTHREAD MUTEXES - We support pthread mutexes locally as they're important to
+// support thread-safe libc operations
+// --------------------------
 
 WAVM_DEFINE_INTRINSIC_FUNCTION(env,
                                "pthread_mutex_init",
                                I32,
                                pthread_mutex_init,
-                               I32 a,
-                               I32 b)
+                               I32 mx,
+                               I32 attr)
 {
-    // faabric::util::getLogger()->trace("S - pthread_mutex_init {} {}", a, b);
+    faabric::util::getLogger()->debug("S - pthread_mutex_init {} {}", mx, attr);
+    threads::getMutexManager().createMutex(mx);
+    return 0;
+}
 
+WAVM_DEFINE_INTRINSIC_FUNCTION(env,
+                               "pthread_mutex_lock",
+                               I32,
+                               pthread_mutex_lock,
+                               I32 mx)
+{
+    faabric::util::getLogger()->debug("S - pthread_mutex_lock {}", mx);
+    threads::getMutexManager().lockMutex(mx);
     return 0;
 }
 
 WAVM_DEFINE_INTRINSIC_FUNCTION(env,
-                               "pthread_cond_init",
+                               "pthread_mutex_trylock",
                                I32,
-                               pthread_cond_init,
-                               I32 a,
-                               I32 b)
+                               s__pthread_mutex_trylock,
+                               I32 mx)
 {
-    // faabric::util::getLogger()->trace("S - pthread_cond_init {} {}", a, b);
+    faabric::util::getLogger()->debug("S - pthread_mutex_trylock {}", mx);
+    bool success = threads::getMutexManager().tryLockMutex(mx);
+    if (success) {
+        return 0;
+    } else {
+        return EBUSY;
```

Comment:
`pthread_mutex_trylock` has a richer set of error codes, but in our case we can assume it's just busy.